# Lauch Application

In [6]:
import pandas as pd
doc = pd.read_csv('datasets/kc_house_data.csv')
doc.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

# Questions | Answers - lesson one

### 1. Quantas casas estão disponíveis para compra?

In [7]:
doc.shape

(21613, 21)

21.613 casas disponíveis para compra.

### 2. Quantos atributos as casas possuem?

In [8]:
doc.shape

(21613, 21)

As casas possuem 21 atributos.

### 3. Quais são os atributos das casas?

In [9]:
doc.columns.tolist()

['id',
 'date',
 'price',
 'bedrooms',
 'bathrooms',
 'sqft_living',
 'sqft_lot',
 'floors',
 'waterfront',
 'view',
 'condition',
 'grade',
 'sqft_above',
 'sqft_basement',
 'yr_built',
 'yr_renovated',
 'zipcode',
 'lat',
 'long',
 'sqft_living15',
 'sqft_lot15']

### 4. Qual a casa mais cara ( casa com o maior valor de venda )?

In [10]:
doc.sort_values('price', ascending= False)

id             date      price  bedrooms  bathrooms  \
7252   6762700020  20141013T000000  7700000.0         6       8.00   
3914   9808700762  20140611T000000  7062500.0         5       4.50   
9254   9208900037  20140919T000000  6885000.0         6       7.75   
4411   2470100110  20140804T000000  5570000.0         5       5.75   
1448   8907500070  20150413T000000  5350000.0         5       5.00   
...           ...              ...        ...       ...        ...   
8274   3883800011  20141105T000000    82000.0         3       1.00   
16198  3028200080  20150324T000000    81000.0         2       1.00   
465    8658300340  20140523T000000    80000.0         1       0.75   
15293    40000362  20140506T000000    78000.0         2       1.00   
1149   3421079032  20150217T000000    75000.0         1       0.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
7252         12050     27600     2.5           0     3  ...     13   
3914         10040     37325     2.0           1     2  ...     11   
9254          9890     31374     2.0           0     4  ...     13   
4411          9200     35069     2.0           0     0  ...     13   
1448          8000     23985     2.0           0     4  ...     12   
...            ...       ...     ...         ...   ...  ...    ...   
8274           860     10426     1.0           0     0  ...      6   
16198          730      9975     1.0           0     0  ...      5   
465            430      5050     1.0           0     0  ...      4   
15293          780     16344     1.0           0     0  ...      5   
1149           670     43377     1.0           0     0  ...      3   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
7252         8570           3480      1910          1987    98102  47.6298   
3914         7680           2360      1940          2001    98004  47.6500   
9254         8860           1030      2001             0    98039  47.6305   
4411         6200           3000      2001             0    98039  47.6289   
1448         6720           1280      2009             0    98004  47.6232   
...           ...            ...       ...           ...      ...      ...   
8274          860              0      1954             0    98146  47.4987   
16198         730              0      1943             0    98168  47.4808   
465           430              0      1912             0    98014  47.6499   
15293         780              0      1942             0    98168  47.4739   
1149          670              0      1966             0    98022  47.2638   

          long  sqft_living15  sqft_lot15  
7252  -122.323           3940        8800  
3914  -122.214           3930       25449  
9254  -122.240           4540       42730  
4411  -122.233           3560       24345  
1448  -122.220           4600       21750  
...        ...            ...         ...  
8274  -122.341           1140       11250  
16198 -122.315            860        9000  
465   -121.909           1200        7500  
15293 -122.280           1700       10387  
1149  -121.906           1160       42882  

[21613 rows x 21 columns]

A casa mais caro id: 6762700020, valor: 7700000.0$

### 5. Qual a casa com o maior número de quartos?

In [11]:
doc.sort_values('bedrooms', ascending= False)

id             date      price  bedrooms  bathrooms  \
15870  2402100895  20140625T000000   640000.0        33       1.75   
8757   1773100755  20140821T000000   520000.0        11       3.00   
15161  5566100170  20141029T000000   650000.0        10       2.00   
13314   627300145  20140814T000000  1148000.0        10       5.25   
19254  8812401450  20141229T000000   660000.0        10       3.00   
...           ...              ...        ...       ...        ...   
19452  3980300371  20140926T000000   142000.0         0       0.00   
8484   2310060040  20140925T000000   240000.0         0       2.50   
875    6306400140  20140612T000000  1095000.0         0       0.00   
8477   2569500210  20141117T000000   339950.0         0       2.50   
9773   3374500520  20150429T000000   355000.0         0       0.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15870         1620      6000     1.0           0     0  ...      7   
8757          3000      4960     2.0           0     0  ...      7   
15161         3610     11914     2.0           0     0  ...      7   
13314         4590     10920     1.0           0     2  ...      9   
19254         2920      3745     2.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
19452          290     20875     1.0           0     0  ...      1   
8484          1810      5669     2.0           0     0  ...      7   
875           3064      4764     3.5           0     2  ...      7   
8477          2290      8319     2.0           0     0  ...      8   
9773          2460      8049     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15870        1040            580      1947             0    98103  47.6878   
8757         2400            600      1918          1999    98106  47.5560   
15161        3010            600      1958             0    98006  47.5705   
13314        2500           2090      2008             0    98004  47.5861   
19254        1860           1060      1913             0    98105  47.6635   
...           ...            ...       ...           ...      ...      ...   
19452         290              0      1963             0    98024  47.5308   
8484         1810              0      2003             0    98038  47.3493   
875          3064              0      1990             0    98102  47.6362   
8477         2290              0      1985             0    98042  47.3473   
9773         2460              0      1990             0    98031  47.4095   

          long  sqft_living15  sqft_lot15  
15870 -122.331           1330        4700  
8757  -122.363           1420        4960  
15161 -122.175           2040       11914  
13314 -122.113           2730       10400  
19254 -122.320           1810        3745  
...        ...            ...         ...  
19452 -121.888           1620       22850  
8484  -122.053           1810        5685  
875   -122.322           2360        4000  
8477  -122.151           2500        8751  
9773  -122.168           2520        8050  

[21613 rows x 21 columns]

A casa com o id: 2402100895 tem o maior número de quartos. Total de 33 quartos.

### 6. Qual a soma total de quartos do conjunto de dados?

In [12]:
doc.bedrooms.sum()

72854

A soma total do conjunto de dados é: 72.854 quartos.

### 7. Quantas casas possuem 2 banheiros?

In [13]:
doc[doc['bathrooms'] == 2].shape

(1930, 21)

1930 casas possuem dois banheiros.

### 8. Qual o preço médio de todas as casas no conjunto de dados?

In [14]:
print(f'{doc.price.mean():0.2f}')

540088.14


O preço médio das casas no conjunto de dados é: 540088.14$

### 9. Qual o preço médio de casas com 2 banheiros?

In [15]:
list_doc_1 = doc.loc[doc.bathrooms == 2]
print(f'{list_home_1.price.mean():.2f}')

NameError: name 'list_home_1' is not defined

O preço médio das casas com dois banheiro é: 457889.72$

### 10. Qual o preço mínimo entre as casas com 3 quartos?

In [ ]:
list_doc_2 = doc.loc[doc.bedrooms == 3]
list_doc_2.price.min()

82.000.00$ é o menor preço das casas com três quartos.

### 11. Quantas casas possuem mais de 300 metros quadrados na sala de estar?

In [ ]:
# OBS: 1 pé quadrado = 0.092
doc.m2_living = doc.sqft_living * 0.092
doc[doc['m2_living']> 300].shape

2141 casas possuem mais de 300m².

### 12. Quantas casas tem mais de 2 andares?

In [ ]:
doc[doc['floors'] > 2].shape

782 casas tem mais de dois andares

### 13. Quantas casas tem vista para o mar?

In [ ]:
doc[doc['waterfront'] > 0].shape

163 casas tem vista para o mar.

### 14. Das casas com vista para o mar, quantas tem 3 quartos?

In [ ]:
list_doc_3 = doc[doc['waterfront'] > 0]
list_doc_3[list_doc_3['bedrooms'] > 3].shape

63 casas com vista para o mar tem três quartos.

### 15. Das casas com mais de 300 metros quadrados de sala de estar, quantas tem mais de 2 banheiros?

In [16]:
# list_doc_4 = doc[doc['m2_living'] > 300]
# list_doc_4[list_doc_4['bathrooms'] > 2].shape
doc[( doc['m2_living'] > 300) & (doc['bathrooms'] > 2)].shape

KeyError: 'm2_living'

2088 casas com mais de 300m² possuem mais de dois banheiros.